tests/commander.js

```diff
@@ -0,0 +1,180 @@
+            //{cmd: "choice before #form/group/select/item", type: "Choice", index: 2, indent: "    "},
+            //{cmd: "Choice after #form/group/select/item", type: "Choice", index: 3, indent: "    "},
+            {cmd: "choice", type: "Choice", index: 3, indent: "    ", select: "group/select"},
+            {cmd: "Choice in #form/group/select", type: "Choice", index: 3, indent: "    "},
+            {cmd: "Choice first in #form/group/select", type: "Choice", index: 2, indent: "    "},
+        ], function (args) {
+            it("should add a " + args.type + " question with '" + args.cmd + "'", function () {
+                util.paste([
+                    ["id", "type", "labelItext:en-default"],
+                    ["/group", "Group", "group"],
+                    ["/group/select", "Select", "select"],
+                    ["/group/select/item", "Choice", "item"],
+                    ["/text", "Text", "text"],
+                ]);
+                if (args.select) {
+                    util.clickQuestion(args.select);
+                }
+                var result = commander.doCommand(args.cmd, vellum);
+                assert.isOk(result, "question not added");
+                $("[name=property-nodeID]").val("new").change();
+                vellum.ensureCurrentMugIsSaved();
+                var items = [
+                    "group",
+                    "  select",
+                    "    item",
+                    "text",
+                ];
+                if (args.hasOwnProperty("index")) {
+                    items.splice(args.index, 0, (args.indent || "") + "new");
+                } else {
+                    items.push("new");
+                }
+                util.assertJSTreeState.apply(null, items);
+                assert.equal(result.__className, args.type);
+            });
+        });
+
+        _.each([
+            "#form/group",
+            "#form/group/select",
+            "#form/text",
+        ], function (path) {
+            it("should select " + path, function () {
+                util.paste([
+                    ["id", "type", "labelItext:en-default"],
+                    ["/group", "Group", "group"],
+                    ["/group/select", "Select", "select"],
+                    ["/group/select/item", "Choice", "item"],
+                    ["/text", "Text", "text"],
+                ]);
+                var result = commander.doCommand(path, vellum);
+                assert.isOk(result, "question not selected");
+                var mug = vellum.getCurrentlySelectedMug();
+                assert.equal(mug.hashtagPath, path);
+            });
+        });
+
+        // TODO decide if : is a good prefix for commands other than add or select
+        // TODO fix selection after delete
+        //  (two nodes selected in tree after delete non-adjacent to current)
+        // TODO delete multiple
+        //_.each([
+        //    ":delete",
```

Comment:
Indeed! I think we could eventually make most mouse interactions accessible through the commander. The other low hanging fruit is menu items, but don't want to get too carried away before I release something useful.

One thing I want to think about before doing an initial release (forgot to mention this when I made the PR) is whether we want to put commands like "Delete" and other menu items in the same namespace (autocomplete menu) as question types, or whether we should use a prefix character (like ":", but I think that's kinda ugly) or whether we should add a command specifically for adding questions? In the latter case, the add question command displayed in the screenshot at the top of this PR would change to "add Text after #group". That's nice for separating the command namespaces, but I'm not terribly fond of that idea because it means more typing to add a question, which is what I think people will use the commander for most of the time. Thoughts welcome.